### PR TITLE
Allow attributes before function arguments

### PR DESCRIPTION
**Pull request by jyn514**
This adds support for function calls of the form:

```rust
  f(
    #[attr(...)] 1.2,
    #[attr_one(...)]
    #[attr_two(...)]
    1.5,
    ... etc ...
  )
```

Closes https://github.com/rust-analyzer/rust-analyzer/issues/2801

### CODE REVIEW

**Comment by jyn514** [comment]
@edwin0cheng mentioned on Zulip that the ATTR should go inside the literal node, not above it. However, I don't know ahead of time whether the node is a literal or not, so I don't know what kind to pass to `complete()`. I added a new `ARG` node, but that caused a bunch of pre-existing tests to fail. The code is here if you want to check it out: https://github.com/jyn514/rust-analyzer/tree/arg-node

**Comment by matklad** [comment]
We have code to push attributes inside the following node in `stmt` function (see this comment https://github.com/rust-analyzer/rust-analyzer/pull/2827). 

We don't have to tackle this from the start, it's ok to just parse attributes *somehow*, and then nest them properly in a follow up RP

File: crates/ra_parser/src/grammar/expressions.rs [hunk 1]
```diff
@@ -535,11 +535,22 @@ fn cast_expr(p: &mut Parser, lhs: CompletedMarker) -> CompletedMarker {
...
+// test arg_list
+// fn assert_float(s: &str, n: f64) {}
+// fn foo() {
+//     assert_float(
+//         "1.797693134862315708e+308L",
+//         #[allow(clippy::excessive_precision)]
+//         #[allow(dead_code)]
+//         1.797_693_134_862_315_730_8e+308,
+//     );
+// }
```
Comment (matklad) [inline crates/ra_parser/src/grammar/expressions.rs]: Inline tests should be minimal. Their purpose is to elucidate how the parsing code works exactly, and not exhaustively tests it.

So, I suggest adding

```
test arg_with_attr
fn main() {
    foo(#[attr] 92)
}
```

right above the `attributes::outer_attributes(p);` line

Comment (jyn514) [reply crates/ra_parser/src/grammar/expressions.rs]: Ok, I can make that change. Is there another place for exhaustive tests?

Comment (matklad) [reply crates/ra_parser/src/grammar/expressions.rs]: I think that this test should be enough, but you can drop additional tests to `test_data/parser/ok`. 

Let's also move the test to the line `547`, so it's close to the code it exercises.

**Comment by jyn514** [comment]
This is ready for review.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**